### ISSUE TITLE
Sending SIGUSR1 when redshift module is active disables it instead of refreshing i3pystatus

### ISSUE THREAD

**Issue by m45t3r** [enkore/i3pystatus#482]
Both i3pystatus and redshift use SIGUSR1 as a special signal. When you try to send SIGUSR1 to i3pystatus process, redshift captures the signal, activating/deactivating inhibit, since the thread inherits the signal handler.

I don't know a good way to fix this for now. We could use `multiprocessing` instead of `threading`, since in this case the process handler is maintained with parent instead of child, however in this case the module will have to be rewritten to allow communication between the two process. Another way would be to rewrite the module to implement the process parsing ourselves, removing the GLib dependency and allowing us to choose which signal to ignore.

Any suggestion are welcome.

**Comment by richese** [comment]
This call added in 3.3 might be usefull: [signal.pthread_sigmask](https://docs.python.org/3/library/signal.html#signal.pthread_sigmask). It can block a thread from receiving certain signals.

Something like this could work if [redshift:73](https://github.com/enkore/i3pystatus/blob/115bd4d34f76bd991ad2ec7a150794a8b8da2e12/i3pystatus/redshift.py#L73) is replaced  with a custom callback.
```python
def thread_callback(self):
    signal.pthread_sigmask(signal.SIG_BLOCK, [signal.SIGUSR1])
    Gtk.main()
```

**Comment by m45t3r** [comment]
@richese I will try this one, thanks.

BTW, any notice about issue #430?

**Comment by m45t3r** [comment]
@richese Does not work. I tried to block all signals with `range(1, signal.NSIG)`, or using `signal.SIG_SETMASK` too, without success.

**Comment by richese** [comment]
Well, it was worth a try.
The USR1 signal handler is added in [constructor]( https://github.com/jonls/redshift/blob/639432fcfa3884c1e2169911c581120b7a7981f8/src/redshift-gtk/statusicon.py#L115) of RedshiftController and I have no idea how to remove or mask it in GLib without the return value from [GLib.unix_signal_add()](http://lazka.github.io/pgi-docs/#GLib-2.0/functions.html#GLib.unix_signal_add).

**Comment by m45t3r** [comment]
@richese I will probably rewrite this module, now I understand what ReshiftController does so it is doable. I think it is possible to implement using pure Python without any additional dependencies except redshift binary itself.

**Comment by m45t3r** [comment]
Should be fixed in #484.
